pallets/portfolio/src/lib.rs

```diff
@@ -0,0 +1,252 @@
+            to_num: Option<PortfolioNumber>,
+            ticker: Ticker,
+            amount: T::Balance
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            ensure!(from_num != to_num, Error::<T>::DestinationIsSamePortfolio);
+            if let Some(from_num) = from_num {
+                ensure!(Self::portfolios(&did, from_num).is_some(), Error::<T>::PortfolioDoesNotExist);
+            }
+            if let Some(to_num) = to_num {
+                ensure!(Self::portfolios(&did, to_num).is_some(), Error::<T>::PortfolioDoesNotExist);
+            }
+            let from_portfolio_id = from_num
+                .and_then(|num| Some(PortfolioId::user_portfolio(did, num)))
+                .unwrap_or_else(|| PortfolioId::default_portfolio(did));
+            let to_portfolio_id = to_num
+                .and_then(|num| Some(PortfolioId::user_portfolio(did, num)))
+                .unwrap_or_else(|| PortfolioId::default_portfolio(did));
+            let from_balance = Self::portfolio_asset_balances(&from_portfolio_id, &ticker);
+            ensure!(from_balance >= amount, Error::<T>::InsufficientPortfolioBalance);
+            <PortfolioAssetBalances<T>>::insert(
+                &from_portfolio_id,
+                &ticker,
+                from_balance - amount
+            );
+            let to_balance = Self::portfolio_asset_balances(&to_portfolio_id, &ticker);
+            <PortfolioAssetBalances<T>>::insert(
+                &to_portfolio_id,
+                &ticker,
+                to_balance.saturating_add(amount)
+            );
+           Self::deposit_event(RawEvent::MovedBetweenPortfolios(
+                did,
+                from_num,
+                to_num,
+                ticker,
+                amount
+            ));
+            Ok(())
+        }
+
+        /// Renames a non-default portfolio.
+        #[weight = 500_000]
+        pub fn rename_portfolio(
+            origin,
+            num: PortfolioNumber,
+            to_name: PortfolioName,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            ensure!(Self::portfolios(&did, &num).is_some(), Error::<T>::PortfolioDoesNotExist);
+            let name_uniq = <Portfolios>::iter_prefix(&did).all(|n| n.1 != to_name);
+            ensure!(name_uniq, Error::<T>::PortfolioNameAlreadyInUse);
+            <Portfolios>::mutate(&did, &num, |p| *p = Some(to_name.clone()));
+            Self::deposit_event(RawEvent::PortfolioRenamed(
+                did,
+                num,
+                to_name,
+            ));
+            Ok(())
+        }
+
+        /// Emits an event containing all non-default portfolio numbers and names of a given DID.
+        #[weight = 500_000]
+        pub fn get_portfolios(origin) -> DispatchResult {
```

Comment:
It can be an extra RPC or even I think it can be directly available by using `iter_prefix_values()` helper function of double_map for the UI team.

Reply:
There is such an RPC method already. See `rpc_get_portfolios` below.

Reply:
Then there is no need for this dispatchable.  @vkomenda

Reply:
I removed this dispatchable. It's however still a question whether we need the RPC method implemented by `rpc_get_portfolios` in this module.

Cc @adamdossa for comments.

Reply:
The rpc method looks good for now.